File: LibraryManagement.API/DTOs/Response/LoanResponse.cs
namespace LibraryManagement.API.DTOs.Response;

public class LoanResponse
{
    public int Id { get; set; }

    public int BookId { get; set; }

    public string BookTitle { get; set; } = string.Empty;

    public int MemberId { get; set; }

    public string MemberFullName { get; set; } = string.Empty;

    public DateTime LoanDate { get; set; }

    public DateTime DueDate { get; set; }

    public DateTime? ReturnDate { get; set; }

    public string Status { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }
}

File: LibraryManagement.Frontend/src/app/core/models/response/loan-response.ts
export interface LoanResponse {
  id: number;
  bookId: number;
  bookTitle: string;
  memberId: number;
  memberFullName: string;
  loanDate: string;
  dueDate: string;
  returnDate?: string;
  status: string;
  createdAt: string;
  updatedAt?: string;
}